### DISCUSSION TITLE
How swc compilers works?

### DISCUSSION THREAD

**Discussion by leamandeep** [swc-project/swc#7605] [General]
Hi swc team! I'm keen interested to know how swc compiler works under the hood. How it's different from Babel and how it managed to compile other javascript framework's code and even it even compiles TS code. how all compilations are doing in Rust? I'm recently started learning compiler design and also a web developer. Bit curious about how swc works.

**Comment by kwonoj** [comment]
Would like to answer, however question is too vague to explain specifics. The way to start is probably reading the codes in the repo, there are couple of pkgs like parser, transform which is directly related to the compiler pipeline which can answer your question in most cases.

**Reply by leamandeep** [reply]
@kwonoj I was reading swc's code and I have a few questions in my mind. If I'm right rust is doing all compilation, bundling, and all the stuff. How rust is doing all that stuff after it's not javascript? how swc manages the backward compatibility of js? I found some Babel packages inside the code. These two questions are wandering in my mind.
